wallet-api/src/Wallet/Graph.hs

```diff
@@ -0,0 +1,99 @@
+{-# LANGUAGE DeriveAnyClass       #-}
+{-# LANGUAGE DeriveGeneric        #-}
+{-# LANGUAGE OverloadedStrings    #-}
+{-# LANGUAGE RecordWildCards      #-}
+{-# OPTIONS_GHC -fno-warn-incomplete-uni-patterns #-}
+
+-- | Support for visualisation of a mockchain as a graph.
+module Wallet.Graph
+  ( txnFlows
+  , graph
+  , FlowGraph
+  , FlowLink
+  , TxRef
+  , UtxOwner
+  ) where
+
+import           Data.Aeson.Types  (ToJSON, toJSON)
+import           Data.Foldable     (fold)
+import           Data.List         (nub)
+import qualified Data.Map          as Map
+import qualified Data.Set          as Set
+import qualified Data.Text         as Text
+import           GHC.Generics      (Generic)
+import           Wallet.UTXO.Types (Blockchain, PubKey, TxId', TxOut (TxOut), TxOutRef (TxOutRef),
+                                    TxOutType (PayToPubKey, PayToScript), getTxId, hashTx, out, txInRef, txInputs,
+                                    txOutRefId, txOutType, txOutValue, unspentOutputs)
+
+-- | Owner of unspent funds
+data UtxOwner
```

Comment:
@krisajenkins The `UtxOwner` field of a `FlowLink` can be used for colouring the diagrams. In my version it was the edges in the sankey diagram that were coloured, but here it is the nodes /transactions. Since a transaction can have multiple inputs it should really be displayed like a stacked bar chart with colours indicating how much each `UtxOwner` contributes to the txn. (Or we could just colour the edges instead of the nodes)

And the bars in the funds distribution bar chart should use the same colours.

And it would be really cool if the "Wallet card", where the initial funds of a wallet are specified, could have an indicator of its colour in the diagram.

That is all :)

Reply:
This can be done in another PR though, we should get this one merged soon